big-o/big-o-notation.md

```diff
@@ -0,0 +1,103 @@
+
+These complexities are ordered! We can describe their order in all of these ways:
+
+- going from least complex to most complex
+- going from fastest to slowest in time complexity
+- going from least operations to most operations in time complexity
+- going from most space efficent to least space efficent in space complexity
+- going from slower-growing to faster-growing
+
+| Complexity       | Name        | Definition                                                                                                                                                                                                                                                                                    |
+| ---------------- | ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| O(1)             | Constant    | The algorithm which will take the same amount of time to execute regardless of the size of the input.                                                                                                                                                                                         |
+| O(log n)         | Logarithmic | The algorithm which will grow in time or space complexity proportional to the **log** of the input size. Logorithmic algorithms increase **very** slowly as the size of the input increases. They usually involve an algorithm which excludes 1/2 of the input with each iteration of a loop. |
+| O(n)             | Linear      | The algorithm will grow in time or space directly proprotional to the input size. The complexity increases at the same rate that the input increases.                                                                                                                                         |
+| O(n log n)       | Log Linear  | A term used to describe an algorithm which will grow in time or space complexity proportional to the n log n of the input size.                                                                                                                                                               |
+| O(n<sup>2</sup>) | Quadratic   | The algorithm which will have a runtime or memory usage proportional to the size of the input squared. This often involves 2 nested loops.                                                                                                                                                    |
+| O(2<sup>n</sup>) | Exponential | The algorithm's complexity doubles each time the input size increases by one.                                                                                                                                                                                                                 |
+
+### !callout-info
```

Comment:
Do these work on Github's Markdown pages? On the rich PR viewer, they didn't work for me.

Reply:
Woops! Sorry, Audrey and I should have given context: A lot of the markdown we're using is for our LMS (learning management system).... aka we supply the markdown, and they have their own markdown to make it ~fancy~ when the students see it.

For fun details: `### !callout-info` is their tag for making a little "callout box." This paragraph will be in a blue box, visually distinct from the rest of the text, feeling like an "aside"